static/js/tab_bar.js

```diff
@@ -69,12 +76,44 @@ function build_tab_bar(filter) {
     } else {
         const tab_bar_data = make_tab_data(filter);
         display_tab_bar(tab_bar_data);
+
         $(".search_closed").on("click", function (e) {
             exports.open_search_bar_and_close_narrow_description();
             search.initiate_search();
             e.preventDefault();
             e.stopPropagation();
         });
+
+        $("#tab_list span:nth-last-child(2)").on("click", function (e) {
+            exports.open_search_bar_and_close_narrow_description();
+            search.initiate_search();
+            e.preventDefault();
+            e.stopPropagation();
+        });
+
+        // Hacky way of protecting the behaviour of links via preventDefault
+        // and stopPropagation
+        $(".narrow_description > a").on("click", function (e) {
+            window.location.href = e.target.href;
+            e.preventDefault();
+            e.stopPropagation();
+        });
```

Comment:
I'm merging this because I think it'll be better to be working on top of this for further testing, but this is a messy/hacky way to implement this.  For links inside the block, we can just do an `if` statement inside the clock handler to decide whether to initiate search (critically, not doing preventDefault if it's a click on an `a` tag), rather than layering multiple overrides on top of each other.